platform/android/MapboxGLAndroidSDKTestApp/src/androidTest/java/com/mapbox/mapboxsdk/testapp/activity/BaseTest.java

```diff
@@ -1,116 +1,113 @@
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import static junit.framework.TestCase.assertNotNull;
+import static junit.framework.TestCase.assertTrue;
 
 /**
  * Base class for all Activity test hooking into an existing Activity that will load style.
  */
-public abstract class BaseTest {
+public abstract class BaseTest implements MapView.OnDidFinishLoadingStyleListener {
+
+  private static final int WAIT_TIMEOUT = 30; //seconds
+  private static final int MAX_TIMEOUT_FAILURE = 3;
+  private static int timeoutCount;
 
   @Rule
-  public ActivityTestRule<Activity> rule = new ActivityTestRule<>(getActivityClass());
+  public ActivityTestRule rule = new ActivityTestRule<>(getActivityClass());
 
   @Rule
-  public TestName testNameRule = new TestName();
+  public TestName testName = new TestName();
 
   protected MapboxMap mapboxMap;
-  protected MapboxIdlingResource idlingResource;
+  protected MapView mapView;
+  private final CountDownLatch latch = new CountDownLatch(1);
 
   @Before
   public void beforeTest() {
-    try {
-      Timber.e(String.format(
-        "%s - %s - %s",
-        getClass().getSimpleName(),
-        testNameRule.getMethodName(),
-        "@Before test: register idle resource"
-      ));
-      idlingResource = (MapboxIdlingResource) generateIdlingResource();
-      IdlingRegistry.getInstance().register(idlingResource);
-      Espresso.onIdle();
-      mapboxMap = idlingResource.getMapboxMap();
-    } catch (IdlingResourceTimeoutException idlingResourceTimeoutException) {
-      throw new RuntimeException(
-        String.format(
-          "Could not start %s test for %s.",
-          testNameRule.getMethodName(),
-          getActivityClass().getSimpleName()
-        )
-      );
-    }
+    initialiseMap();
+    holdTestRunnerForStyleLoad();
   }
 
-  protected IdlingResource generateIdlingResource() {
-    return new FinishLoadingStyleIdlingResource(rule.getActivity());
+  private void initialiseMap() {
+    try {
+      rule.runOnUiThread(() -> {
+        mapView = rule.getActivity().findViewById(R.id.mapView);
+        mapView.addOnDidFinishLoadingStyleListener(this);
+        mapView.getMapAsync(this::initMap);
+      });
+    } catch (Throwable throwable) {
+      throwable.printStackTrace();
+    }
   }
 
-  protected void validateTestSetup() {
-    if (!Mapbox.isConnected()) {
-      Timber.e("Not connected to the internet while running test");
+  private void holdTestRunnerForStyleLoad() {
+    boolean interrupted;
+    try {
+      interrupted = latch.await(WAIT_TIMEOUT, TimeUnit.SECONDS);
+    } catch (InterruptedException ignore) {
+      interrupted = true;
     }
 
-    checkViewIsDisplayed(R.id.mapView);
-    Assert.assertNotNull(mapboxMap);
-  }
+    if (!interrupted) {
+      Timber.e("Timeout occurred for %s", testName.getMethodName());
 
-  protected MapboxMap getMapboxMap() {
-    return mapboxMap;
+      // update global timeout counter
+      timeoutCount++;
```

Comment:
note that this is temporary measure. 
The idea was that it would fail calling `validateTestSetup`:

https://github.com/mapbox/mapbox-gl-native/pull/14051/files/2c45003f599b85279aad264bd7d0d5b9e2c3e334#diff-0baceb1a285e95bf6a17565e064d9052R96

and show what wasn't initialized. For more clarity in code I should call validateTestSetup here.

Reply:
I removed the timecount setup and now calling validate test setup directly